provider/cloudflare/cloudflare_regional.go

```diff
@@ -72,89 +85,93 @@ func updateDataLocalizationRegionalHostnameParams(rhc regionalHostnameChange) cl
-					}
-					continue
-				}
-				failedChange = true
-				log.WithFields(logFields).Errorf("failed to update regional hostname: %v", err)
-			}
-		case cloudFlareDelete:
-			log.WithFields(logFields).Debug("Deleting regional hostname")
-			if p.DryRun {
-				continue
-			}
-			err := p.Client.DeleteDataLocalizationRegionalHostname(ctx, resourceContainer, rhChange.Hostname)
-			if err != nil {
-				var apiErr *cloudflare.Error
-				if errors.As(err, &apiErr) && apiErr.StatusCode == http.StatusNotFound {
-					log.WithFields(logFields).Debug("Regional hostname does not exists, nothing to do")
-					continue
-				}
-				failedChange = true
-				log.WithFields(logFields).Errorf("failed to delete regional hostname: %v", err)
-			}
+		if !p.submitRegionalHostnameChange(ctx, rhChange, resourceContainer) {
+			failedChange = true
 		}
 	}
 
 	return !failedChange
 }
 
+// submitRegionalHostnameChange applies a single regional hostname change, returns false if it fails
+func (p *CloudFlareProvider) submitRegionalHostnameChange(ctx context.Context, rhChange regionalHostnameChange, resourceContainer *cloudflare.ResourceContainer) bool {
+	logFields := log.Fields{
+		"hostname":   rhChange.Hostname,
+		"region_key": rhChange.RegionKey,
+		"action":     rhChange.action,
+		"zone":       resourceContainer.Identifier,
+	}
+	switch rhChange.action {
+	case cloudFlareCreate:
+		log.WithFields(logFields).Debug("Creating regional hostname")
+		if p.DryRun {
+			return true
+		}
+		regionalHostnameParam := createDataLocalizationRegionalHostnameParams(rhChange)
+		err := p.Client.CreateDataLocalizationRegionalHostname(ctx, resourceContainer, regionalHostnameParam)
+		if err != nil {
+			log.WithFields(logFields).Errorf("failed to create regional hostname: %v", err)
+			return false
+		}
+	case cloudFlareUpdate:
+		log.WithFields(logFields).Debug("Updating regional hostname")
+		if p.DryRun {
+			return true
+		}
+		regionalHostnameParam := updateDataLocalizationRegionalHostnameParams(rhChange)
+		err := p.Client.UpdateDataLocalizationRegionalHostname(ctx, resourceContainer, regionalHostnameParam)
+		if err != nil {
+			log.WithFields(logFields).Errorf("failed to update regional hostname: %v", err)
+			return false
+		}
+	case cloudFlareDelete:
+		log.WithFields(logFields).Debug("Deleting regional hostname")
+		if p.DryRun {
+			return true
+		}
+		err := p.Client.DeleteDataLocalizationRegionalHostname(ctx, resourceContainer, rhChange.Hostname)
+		if err != nil {
+			log.WithFields(logFields).Errorf("failed to delete regional hostname: %v", err)
+			return false
+		}
+	}
+	return true
+}
+
+func (p *CloudFlareProvider) listDataLocalisationRegionalHostnames(ctx context.Context, resourceContainer *cloudflare.ResourceContainer) (RegionalHostnamesMap, error) {
+	rhs, err := p.Client.ListDataLocalizationRegionalHostnames(ctx, resourceContainer, cloudflare.ListDataLocalizationRegionalHostnamesParams{})
+	if err != nil {
+		var apiErr *cloudflare.Error
```

Comment:
shell we use this method here https://github.com/kubernetes-sigs/external-dns/blob/17fa4b4e7aa410779d3c8db0745e75e98f5d98a9/provider/cloudflare/cloudflare.go#L274?